website/docs/rbs-support.md

```diff
@@ -34,7 +35,179 @@ def foo(x)
+
+Sure, there are similarities: both syntaxes offer a way to express union types,
+for example. But there are also semantic differences, and RBS syntax reflects
+these differences. Things which are possible to express in RBS syntax have no
+analogue in Sorbet and vice versa. Some examples:
+
+- RBS supports duck typing, but
+  [Sorbet does not](faq.md#can-i-use-sorbet-for-duck-typed-code), by design
+- Sorbet allows singleton classes to be generic (e.g.
+  [type_template](generics.md#type_member--type_template)). RBS has no such
+  concept.
+- Sorbet treats Ruby's `::Class` as a generic class, allowing sophisticated
+  class-level metaprogramming. RBS does not have syntax to represent this.
+
+Because of these differences, it's reasonable to assume that a codebase wishing
+to take full advantage of Sorbet's unique features will eventually need to have
+annotations that use `sig` syntax. The moment a method's annotation needs to use
+Sorbet-only syntax, the entire annotation needs to get rewritten, adding
+friction to the development experience.
+
+#### Sorbet has minimal influence over the evolution of RBS syntax
+
+Sorbet continues to evolve its type syntax. For example, `T.anything`,
+`T::Class`, and `has_attached_class!` are additions to Sorbet which arrived 6
+years after Sorbet's inception—Sorbet development is anything but stale!
+
+Sorbet can adapt more quickly to the needs of Sorbet users by building on its
+own syntax. Historically, it has been difficult to advocate to adding RBS
+features that benefit Sorbet specifically.
+
+Our belief is that syntax matters less than semantics; that "what's possible to
+express in the type system" matters more than "how to express it." Time spent
+bikeshedding syntax detracts from meaningful improvements to the type system.
+
+#### IDE integration is more difficult
+
+Sorbet's type syntax doubles as Ruby syntax. Setting aside other benefits of
+reusing Ruby syntax for type annotations (e.g., [runtime checking](runtime.md),
+no need to transpile, seamless integration with linters, etc.), Sorbet's syntax
+integrates easily with existing editors.
+
+There are a lot of features that come for free by reusing Ruby syntax:
+
+- Syntax highlighting
+
+  There are no plans to offer highlighting for RBS comment-based type
+  annotations. They will be monochromatic, highlighted like comments
+  (third-party extensions may offer such functionality).
+
+- Syntax error tolerance and recovery
+
+  The RBS comment parser does not recover from syntax errors. If there's a
+  syntax error in an RBS comment, Sorbet will ignore the entire `sig` until the
+  error is fixed.
+
+- Autocompletion
+
+  Without error tolerance, autocompletion support is unreliable.
+
+In fairness, these are technical, implementation considerations, and thus could
+hope to improve one day. But they remain problems today.
+
+#### No support for inline type annotations
+
+At time of writing, there is no support for inline type annotations and
```

Comment:
We're working on it but you're right that the type assertion will need to be on its own line.